library/leon/io/StdIn.scala

```diff
@@ -16,38 +16,38 @@ object StdIn {
   @isabelle.noBody()
   def readInt(implicit state: State): Int = {
     state.seed += 1
-    nativeReadInt
+    nativeReadInt(state.seed)
   }
 
   @library
   @extern
-  private def nativeReadInt(implicit state: State): Int = {
+  private def nativeReadInt(seed: BigInt): Int = {
```

Comment:
Ho, I see! Actually, I realised just now that's not an issue for GenC in this case because I'm dropping the `nativeReadInt` anyway (the manual translation to C is directly plugged in `readInt`).